catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/GeoPdfParser.java

```diff
@@ -0,0 +1,292 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input.pdf;
+
+import java.awt.geom.AffineTransform;
+import java.awt.geom.Point2D;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pdfbox.cos.COSArray;
+import org.apache.pdfbox.cos.COSBase;
+import org.apache.pdfbox.cos.COSBoolean;
+import org.apache.pdfbox.cos.COSDictionary;
+import org.apache.pdfbox.cos.COSDocument;
+import org.apache.pdfbox.cos.COSFloat;
+import org.apache.pdfbox.cos.COSInteger;
+import org.apache.pdfbox.cos.COSName;
+import org.apache.pdfbox.cos.COSNull;
+import org.apache.pdfbox.cos.COSStream;
+import org.apache.pdfbox.cos.COSString;
+import org.apache.pdfbox.cos.ICOSVisitor;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GeoPdfParser {
+
+    public static final String GEOGRAPHIC = "GEOGRAPHIC";
+
+    public static final String LGIDICT = "LGIDict";
+
+    public static final String PROJECTION = "Projection";
+
+    public static final String PROJECTION_TYPE = "ProjectionType";
+
+    public static final String NEATLINE = "Neatline";
+
+    public static final String CTM = "CTM";
+
+    private static final String POLYGON = "POLYGON ((";
+
+    private static final String MULTIPOLYGON = "MULTIPOLYGON (";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GeoPdfParser.class);
+
+    /**
+     * Generates a WKT compliant String from a PDF Document if it contains GeoPDF information.
+     * Currently, only WGS84 Projections are supported (GEOGRAPHIC GeoPDF ProjectionType).
+     *
+     * @param pdfDocument - The PDF document
+     * @return the WKT String
+     * @throws IOException
+     */
+    public String getWktFromPDF(PDDocument pdfDocument) throws IOException {
+        ToDoubleVisitor toDoubleVisitor = new ToDoubleVisitor();
+        LinkedList<String> polygons = new LinkedList<>();
+
+        for (PDPage pdPage : pdfDocument.getPages()) {
```

Comment:
Might as well store the `COSObject` in a local variable since you're calling `pdPage.getCOSObject()` a lot.